app/src/tests/components/contents/PacketGroup/PacketTable.test.tsx

```diff
@@ -0,0 +1,113 @@
+import { SWRConfig } from "swr";
+import { PacketTable } from "../../../../app/components/contents/PacketGroup/PacketTable";
+import { render, screen, waitFor } from "@testing-library/react";
+import { mockPacket, mockPacketGroupResponse } from "../../../mocks";
+import { MemoryRouter, Route, Routes } from "react-router-dom";
+import userEvent from "@testing-library/user-event";
+import { server } from "../../../../msw/server";
+import { rest } from "msw";
+import appConfig from "../../../../config/appConfig";
+import { HttpStatus } from "../../../../lib/types/HttpStatus";
+
+const renderComponent = (parentLoading = false) =>
+  render(
+    <SWRConfig value={{ dedupingInterval: 0 }}>
+      <MemoryRouter initialEntries={[`/${mockPacketGroupResponse.content[0].name}`]}>
+        <Routes>
+          <Route path="/:packetName" element={<PacketTable parentIsLoading={parentLoading} />} />
+        </Routes>
+      </MemoryRouter>
+    </SWRConfig>
+  );
+describe("Packet table", () => {
+  it("should render loading skeleton when parent is loading", async () => {
+    const { container } = renderComponent(true);
+
+    expect(container.querySelector(".animate-pulse")).toBeVisible();
+  });
+
+  it("should render error component when error fetching packets", async () => {
+    server.use(
+      rest.get(`${appConfig.apiUrl()}/packets/${mockPacket.name}`, (req, res, ctx) => {
+        return res(ctx.status(400));
+      })
+    );
+    renderComponent();
+
+    await waitFor(() => {
+      expect(screen.getByText(/error fetching packets/i)).toBeVisible();
+    });
+  });
+
+  it("should render unauthorized when 401 error fetching packets", async () => {
+    server.use(
+      rest.get(`${appConfig.apiUrl()}/packets/${mockPacket.name}`, (req, res, ctx) => {
+        return res(ctx.status(HttpStatus.Unauthorized));
+      })
+    );
+    renderComponent();
+
+    await waitFor(() => {
+      expect(screen.getByText(/unauthorized/i)).toBeVisible();
+    });
+  });
+
+  it("should be able to sort by packet column", async () => {
```

Comment:
This test doesn't test sorting, only that the button icon changes.

Reply:
yeah ive renamed text... because sorting is done by the underlying library (thus not testing) that behaviour